src/main/java/io/cdap/plugin/servicenow/apiclient/ServiceNowTableAPIClientImpl.java

```diff
@@ -375,45 +238,146 @@ public List<Map<String, Object>> fetchTableRecordsRetryableMode(String tableName
    * @return schema for given ServiceNow table
    */
   @Nullable
-  public Schema fetchServiceNowTableSchema(String tableName, FailureCollector collector) {
+  public Schema fetchTableSchema(String tableName, FailureCollector collector) {
+    Schema schema = null;
+    try {
+      schema = fetchTableSchema(tableName);
+    } catch (OAuthProblemException | OAuthSystemException | RuntimeException e) {
+      LOG.error("Error in connection - {}", e);
+      collector.addFailure("Connection failed. Unable to fetch schema for table " + tableName, null);
+    }
+    return schema;
+  }
+
+  @VisibleForTesting
+  public List<SchemaResponse> parseSchemaResponse(String responseBody) {
+    JsonObject jo = gson.fromJson(responseBody, JsonObject.class);
+    JsonArray ja = jo.getAsJsonArray(ServiceNowConstants.RESULT);
+    Type type = new TypeToken<List<SchemaResponse>>() {
+    }.getType();
+    return gson.fromJson(ja, type);
+  }
+
+  /**
+   * Fetches the table schema from ServiceNow
+   *
+   * @param tableName ServiceNow table name for which schema is getting fetched
+   * @return schema for given ServiceNow table
+   * @throws OAuthProblemException
+   * @throws OAuthSystemException
+   */
+  public Schema fetchTableSchema(String tableName) throws OAuthProblemException, OAuthSystemException {
     ServiceNowTableAPIRequestBuilder requestBuilder = new ServiceNowTableAPIRequestBuilder(
       this.conf.getRestApiEndpoint(), tableName, true)
       .setExcludeReferenceLink(true);
 
     RestAPIResponse apiResponse;
+    String accessToken = getAccessToken();
+    requestBuilder.setAuthHeader(accessToken);
+    apiResponse = executeGet(requestBuilder.build());
+    if (!apiResponse.isSuccess()) {
+      LOG.error("Error - {}", getErrorMessage(apiResponse.getResponseBody()));
+      throw new RuntimeException("Error - " + getErrorMessage(apiResponse.getResponseBody()));
+    }
+    List<SchemaResponse> result = parseSchemaResponse(apiResponse.getResponseBody());
+    List<ServiceNowColumn> columns = new ArrayList<>();
+
+    if (result != null && !result.isEmpty()) {
```

Comment:
> if it's null or empty, it looks like it will result in an exception when the schema is created. That will be a more confusing error message than if it is handled directly here instead:
> 
> ```
> if (result == null || result.isEmpty()) {
>   // throw some exception with useful message
> }
> for (SchemaResponse field : result) {
>   ....
> }
> ```
Throwing Exception with a message.

Introduced a new class `SchemaResponse` as per the previous review comment and replaced the current one by `SchemaField` e63c214cf864e664ca24072f21c719c1a777b9c4